pkg/git/WebhookEventService.go

```diff
@@ -0,0 +1,295 @@
+	if err != nil {
+		impl.logger.Errorw("getting error while fetching webhook event parsed data ","err", err)
+		return nil, err
+	}
+
+	return webhookEventParsedData, nil
+}
+
+func (impl WebhookEventServiceImpl) SaveWebhookParsedEventData(webhookEventParsedData *sql.WebhookEventParsedData) error {
+	impl.logger.Debug("saving webhook parsed event data")
+	err := impl.webhookEventRepository.SaveWebhookParsedEventData(webhookEventParsedData)
+	if err != nil {
+		impl.logger.Errorw("error while saving webhook parsed event data ","err", err)
+		return err
+	}
+	return nil
+}
+
+func (impl WebhookEventServiceImpl) UpdateWebhookParsedEventData(webhookEventParsedData *sql.WebhookEventParsedData) error {
+	impl.logger.Debugw("updating webhook parsed event data for id : ", webhookEventParsedData.Id)
+	err := impl.webhookEventRepository.UpdateWebhookParsedEventData(webhookEventParsedData)
+	if err != nil {
+		impl.logger.Errorw("error while updating webhook parsed event data ","err", err)
+		return err
+	}
+	return nil
+}
+
+
+func (impl WebhookEventServiceImpl) MatchCiTriggerConditionAndNotify(event *sql.GitHostWebhookEvent, webhookEventParsedData *sql.WebhookEventParsedData, fullDataMap map[string]string) error {
+
+	repositoryUrl := fullDataMap[WEBHOOK_SELECTOR_REPOSITORY_URL_NAME]
+
+	if len(repositoryUrl) == 0{
+		impl.logger.Warn("repository url is blank. so skipping matching condition")
+		return nil
+	}
+
+	impl.logger.Debug("Checking for auto CI in case of webhook event")
+
+	// get materials by Urls
+	var repoUrls []string
+	repoUrls = append(repoUrls, repositoryUrl)
+	repoUrls = append(repoUrls, fmt.Sprintf("%s%s", repositoryUrl, ".git"))
+	materials, err := impl.materialRepository.FindAllActiveByUrls(repoUrls)
+
+	if err != nil {
+		impl.logger.Errorw("error in fetching active materials", "err", err)
+		return err
+	}
+
+	if len(materials) == 0 {
+		impl.logger.Info("no materials found skipping.")
+		return nil
+	}
+
+	for _, material := range materials {
+		ciPipelineMaterials := material.CiPipelineMaterials
+		if len(ciPipelineMaterials) == 0 {
+			impl.logger.Infow("no ci pipeline, skipping", "id", material.Id, "url", material.Url)
+			continue
+		}
+
+		for _, ciPipelineMaterial := range ciPipelineMaterials {
+
+			// ignore if type does not match
+			if ciPipelineMaterial.Type != sql.SOURCE_TYPE_WEBHOOK {
+				continue
+			}
+
+			//MatchFilter
+			match := impl.MatchFilter(event, fullDataMap, ciPipelineMaterial.Value)
+
+
+			// insert/update mapping into DB
+			err = impl.HandleMaterialWebhookMappingIntoDb(ciPipelineMaterial.Id, webhookEventParsedData.Id, match)
+			if err != nil {
+				impl.logger.Errorw("err in saving mapping", "err", err)
+				return err
+			}
+
+			// if condition is match, then notify for CI
+			if match {
+				impl.NotifyForAutoCi(impl.BuildNotifyCiObject(ciPipelineMaterial, webhookEventParsedData))
```

Comment:
unhandled error

Reply:
Yes, intentionally unhandled. 
For git providers (Github etc..) - this is case of success. we are failing in our internal call, so we are logging here